### ISSUE TITLE
MaxReaders 1 not working for RTSP(S)

### ISSUE THREAD

**Issue by tbnguyen1407** [bluenviron/mediamtx#2246]
## Which version are you using?

v1.0.0

## Which operating system are you using?

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [x] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

MaxReaders config is not enforced properly for RTSP(S). It seems those protocols require 2 readers (even though path.readers only show 1 session). Setting MaxReaders to 1 therefore would not allow any RTSP(S) client connection.

## Describe how to replicate the issue

1. start the server
2. publish path with maxReaders set to 1
3. read RTSP with any client

```
INF [RTSP] [conn 127.0.0.1:62723] closed (maximum reader count reached) 
```

## Did you attach the server logs?

no

## Did you attach a network dump?

no

**Comment by machenmusik** [comment]
Is that with tcp, udp, or both?

**Comment by tbnguyen1407** [comment]
I tested with tcp mediamtx output and tcp client.

```
~ mediamtx.yaml ~
protocols: [tcp]
```

```
$ ffplay -rtsp_transport tcp http://localhost:8554/mystream
```

**Comment by aler9** [comment]
fixed in v1.0.1
